specutils/fitting/fitmodels.py

```diff
@@ -1,20 +1,214 @@
+    line_inds_grouped = _consecutive(pos_inds, stepsize=1)
+
+    if len(line_inds_grouped[0]) > 0:
+        emission_inds = [inds[np.argmax(spectrum.flux.value[inds])] for inds in line_inds_grouped]
+    else:
+        emission_inds = []
+
+    #
+    # Find the absorption lines
+    #
+
+    neg_inds = inds[spectrum.flux.value[inds] < 0]
+    line_inds_grouped = _consecutive(neg_inds, stepsize=1)
+
+    if len(line_inds_grouped[0]) > 0:
+        absorption_inds = [inds[np.argmin(spectrum.flux.value[inds])] for inds in line_inds_grouped]
+    else:
+        absorption_inds = []
+
+    #
+    # Create the QTable to return the lines
+    #
+
+    qtable = QTable()
+    qtable['lines'] = list(itertools.chain(*[spectrum.spectral_axis.value[emission_inds],
+                                             spectrum.spectral_axis.value[absorption_inds]]))*spectrum.spectral_axis.unit
+    qtable['line_type'] = ['emission']*len(emission_inds) + ['absorption']*len(absorption_inds)
+    qtable['index'] = list(itertools.chain(*[emission_inds, absorption_inds]))
+
+    return qtable
+
+
+def find_lines_derivative(spectrum, sigma):
+    """
+    Find the emission and absorption lines in a spectrum. The method
+    here is a very simple threshold based on the standard deviation
+    of the noise.
+
+    Parameters
+    ----------
+    spectrum : Spectrum1D
+        The spectrum object over which the equivalent width will be calculated.
+
+    sigma : float
+        Threshold on line finding.
+
+    Returns
+    -------
+    qtable: `~astropy.table.QTable`
+        Table of emission and absorption lines. Spectral axis values, line type and indices stored.
+    """
+
+    # Take the derivative to find the zero crossings which correspond to
+    # the peaks (positive or negative)
+    kernel = [1, 0, -1]
+    dY = convolve(spectrum.flux, kernel, 'valid')
+
+    # Use sign flipping to determine direction of change
+    S = np.sign(dY)
+    ddS = convolve(S, kernel, 'valid')
+
+    #
+    # Emmision lines
+    #
+
+    # Find all the indices that appear to be part of a +ve peak
+    candidates = np.where(dY > 0)[0] + (len(kernel) - 1)
+    line_inds = sorted(set(candidates).intersection(np.where(ddS == -2)[0] + 1))
+    line_inds = np.array(line_inds)[spectrum.flux[line_inds] > sigma*spectrum.flux.unit]
+
+    # Now group them and find the max highest point.
+    line_inds_grouped = _consecutive(line_inds, stepsize=1)
+
+    if len(line_inds_grouped[0]) > 0:
+        emission_inds = [inds[np.argmax(spectrum.flux[inds])] for inds in line_inds_grouped]
+    else:
+        emission_inds = []
+
+    #
+    # Absorption lines
+    #
+
+    # Find all the indices that appear to be part of a -ve peak
+    candidates = np.where(dY < 0)[0] + (len(kernel) - 1)
+    line_inds = sorted(set(candidates).intersection(np.where(ddS == 2)[0] + 1))
+    line_inds = np.array(line_inds)[spectrum.flux[line_inds] < -sigma*spectrum.flux.unit]
```

Comment:
Why is this called `sigma`? It doesn't seem like its related to the standard deviation... isn't it just a flux threshold (and if so, maybe it should be `flux_threshold`)?  And if so, it should be provided as a `Quantity` so that users can give whatever units they prefer (as long as they are compatible with the `spectrum.flux`)

Reply:
good point. done